File: BookStore/BookStore/wwwroot/js/datatable/book.js
let dataTable = $('#bookTable').DataTable({
    "ajax": {
        "url": "/Admin/Book/GetBooks",
        "type": "GET",
    },
    "columns": [
        { "data": "isbn" },
        { "data": "title" },
        { "data": "author" },
        { "data": "publisher" },
        { "data": "purchasePrice" },
        { "data": "offeringPrice" },
        { "data": "edition" },
        { "data": "publicationYear" },
        { "data": "status" },
        { "data": "quantity" },
        {
            "data": "bookId",
            "render": function (data) {
                return `
                    <a href="/Admin/Book/Index?id=${data}" class="btn btn-success p-2">Edit</a>
                    <button onclick="MyDelete('${data}')" id="toastbtn" class="btn btn-danger p-2">Delete</button>
                `;
            }
        }
    ]
});

function MyDelete(bookId) {
    var toastElList = [].slice.call(document.querySelectorAll('.toast'))
    var toastList = toastElList.map(function (toastEl) {
        return new bootstrap.Toast(toastEl)
    })

    let isConfirmed = confirm("Are you sure to delete?");

    if (isConfirmed) {
        $.ajax({
            url: "/Admin/Book/Delete?id=" + bookId,
            type: "DELETE",
            success: function (data) {
                if (data.success) {
                    document.getElementById("toastHeader").setAttribute("class", "toast-header text-white bg-success");
                    document.getElementById("toastBody").innerHTML = "Book has been successfully deleted";
                    dataTable.ajax.reload();
                }
                else {
                    document.getElementById("toastHeader").setAttribute("class", "toast-header text-white bg-danger");
                    document.getElementById("toastBody").innerHTML = "Error has been occured";
                }

                toastList.forEach(toast => toast.show());                 
            }
        })
    }
}

File: BookStore/BookStore.Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookStore.Models
{
    public class Book
    {
        public Guid BookId { get; set; }
        public string ISBN { get; set; }
        public string Author { get; set; }
        public string Title { get; set; }
        public string Edition { get; set; }
        public int PublicationYear { get; set; }
        public string Publisher { get; set; }
        public double Price { get; set; }
        public double DiscountRate { get; set; }
        public string ImageUrl { get; set; }
        public string Description { get; set; }
        public int Quantity { get; set; }
        public string Status { get; set; }
        public double OfferingPrice { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public List<Cart> Carts { get; set; }
        public List<OrderDetail> OrderDetails { get; set; }
    }
}
